src/main/java/pl/coderslab/projectwybankuj/service/LoanService.java

```diff
@@ -0,0 +1,148 @@
+        return payment.setScale(ROUND_SCALE, RoundingMode.HALF_UP);
+    }
+
+    public BigDecimal calculateServiceCharge(Loan loan, int amount) {
+        BigDecimal serviceCharge = BigDecimal.valueOf(amount).multiply(loan.getServiceCharge().divide(BigDecimal.valueOf(100), CALCULATE_SCALE, RoundingMode.CEILING));
+        return serviceCharge.setScale(ROUND_SCALE, RoundingMode.HALF_UP);
+    }
+
+    public BigDecimal calculateInsurance(Loan loan, int amount) {
+        BigDecimal insurance = BigDecimal.valueOf(amount).multiply(loan.getInsurance().divide(BigDecimal.valueOf(100), CALCULATE_SCALE, RoundingMode.CEILING));
+        return insurance.setScale(ROUND_SCALE, RoundingMode.HALF_UP);
+    }
+
+    public BigDecimal calculateInterestsCost(int amount, int creditPeriod, BigDecimal payment, BigDecimal serviceCharge, BigDecimal insurance) {
+        BigDecimal interests = BigDecimal.valueOf(creditPeriod).multiply(payment).subtract(BigDecimal.valueOf(amount)).subtract(serviceCharge).subtract(insurance);
+        return interests.setScale(ROUND_SCALE, RoundingMode.HALF_UP);
+    }
+
+    public BigDecimal calculateTotalCost(int amount, int creditPeriod, BigDecimal payment) {
+        BigDecimal interests = BigDecimal.valueOf(creditPeriod).multiply(payment).subtract(BigDecimal.valueOf(amount));
+        return interests.setScale(ROUND_SCALE, RoundingMode.HALF_UP);
+    }
+
+    private Map<Loan, BigDecimal> sortByPayment(Map<Loan, BigDecimal> mapToSort) {
+        Map<Loan, BigDecimal> sortedMap =
+                mapToSort.entrySet().stream()
+                        .sorted(Entry.comparingByValue())
+                        .collect(Collectors.toMap(Entry::getKey, Entry::getValue,
+                                (e1, e2) -> e1, LinkedHashMap::new));
+
+        return sortedMap;
+    }
+
+    public List<Loan> chooseOffers(String chooseInsurance, String chooseServiceCharge,
+                                   int amount, int creditPeriod, int age) {
+        List<Loan> loans = new ArrayList<>();
+        if (chooseInsurance.equals("no") && chooseServiceCharge.equals("no")) {
+            loans = loanRepository.findAllByParametersWithoutInsuranceAndServiceCharge(amount, creditPeriod, age);
+        } else if (chooseInsurance.equals("no") && chooseServiceCharge.equals("yes")) {
+            loans = loanRepository.findAllByParametersWithoutInsurance(amount, creditPeriod, age);
+        } else if (chooseServiceCharge.equals("no") && chooseInsurance.equals("yes")) {
+            loans = loanRepository.findAllByParametersWithoutServiceCharge(amount, creditPeriod, age);
+        } else {
+            loans = loanRepository.findAllByParameters(amount, creditPeriod, age);
+        }
+        return loans;
+    }
+
+    public void generatePDF(Loan loan, UserLoan userLoan, BigDecimal payment,
```

Comment:
Czy ta metoda napewno musi być w tym serwisie? Odpowiadając sobie na to pytanie warto spojrzeć czy korzysta z pól tej klasy czy tez nie.